src/main/java/picard/arrays/VcfToAdpc.java

```diff
@@ -0,0 +1,164 @@
+                    "An adpc.bin file is a binary file containing genotyping array intensity data that can be exported " +
+                    "by Illumina's GenomeStudio and Beadstudio analysis tools. The adpc.bin file is used as an input to " +
+                    "<a href='https://genome.sph.umich.edu/wiki/VerifyIDintensity'>VerifyIDintensity</a> a tool for " +
+                    "detecting and estimating sample contamination of Illumina genotyping array data." +
+                    "<h4>Usage example:</h4>" +
+                    "<pre>" +
+                    "java -jar picard.jar VcfToAdpc \\<br />" +
+                    "      VCF=input.vcf \\<br />" +
+                    "      OUTPUT=output.adpc.bin" +
+                    "</pre>";
+
+
+    private final Log log = Log.getInstance(VcfToAdpc.class);
+
+    @Argument(doc = "The Input VCF")
+    public File VCF;
+
+    @Argument(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output (adpc.bin) file to write.")
+    public File OUTPUT;
+
+    @Override
+    protected int doWork() {
+        IOUtil.assertFileIsReadable(VCF);
+        IOUtil.assertFileIsWritable(OUTPUT);
+        VCFFileReader vcfFileReader = new VCFFileReader(VCF, false);
+
+        try (CloseableIterator<VariantContext> variants = vcfFileReader.iterator();
+             final IlluminaAdpcFileWriter adpcFileWriter = new IlluminaAdpcFileWriter(OUTPUT)) {
+
+            final List<IlluminaAdpcFileWriter.Record> adpcRecordList = new ArrayList<>();
+            while (variants.hasNext()) {
+                final VariantContext context = variants.next();
+                final float gcScore = getFloatAttribute(context, InfiniumVcfFields.GC_SCORE);
+
+                for (final Genotype genotype : context.getGenotypes()) {
+                    final IlluminaGenotype illuminaGenotype = getIlluminaGenotype(genotype, context);
+
+                    final short rawXIntensity = getShortAttribute(genotype, InfiniumVcfFields.X);
+                    final short rawYIntensity = getShortAttribute(genotype, InfiniumVcfFields.Y);
+                    final float normalizedXIntensity = getFloatAttribute(genotype, InfiniumVcfFields.NORMX);
+                    final float normalizedYIntensity = getFloatAttribute(genotype, InfiniumVcfFields.NORMY);
+
+                    final IlluminaAdpcFileWriter.Record record = new IlluminaAdpcFileWriter.Record(rawXIntensity, rawYIntensity, normalizedXIntensity, normalizedYIntensity, gcScore, illuminaGenotype);
+                    adpcRecordList.add(record);
+                }
+            }
+            adpcFileWriter.write(adpcRecordList);
+        } catch (Exception e) {
+            log.error(e);
+            return 1;
+        }
+
+        return 0;
+    }
+
+    private IlluminaGenotype getIlluminaGenotype(final Genotype genotype, final VariantContext context) {
+        IlluminaGenotype illuminaGenotype = picard.arrays.illumina.IlluminaGenotype.NN;
```

Comment:
make this final and then error out with a final else below

Reply:
Ok